libdnf/hy-query.c

```diff
@@ -657,9 +657,15 @@ filter_updown(HyQuery q, int downgrade, Map *res)
     int i;
     Map m;
 
-    assert(pool->installed);
     dnf_sack_make_provides_ready(q->sack);
     map_init(&m, pool->nsolvables);
+
+    if (!pool->installed) {
+        map_and(res, &m);
```

Comment:
shouldn't we just `return` before doing `map_init()`?

I'm not sure what this function is supposed to do from quick look, so if you could explain what this change is supposed to do in commit message -- would be great

Reply:
We need that empty map to remove empty query